### PR TITLE
feat(svg-editor): expose vector sub-selection API in path-edit mode (#790)

### LINKED ISSUE #790: @grida/svg-editor: expose initial vertex sub-selection in path-edit mode
## Summary

There's no public way to set the **vertex sub-selection** when entering path-edit (vector) mode. A host can put the editor into content-edit on a `<path>` (`editor.enter_content_edit(id)`), and the HUD renders all vertex knobs — but every knob opens **unselected**, and there's no API to pre-select one (or several).

## Current state

- Vertex sub-selection lives entirely inside the surface's `VectorEditSession` — `select_vertex(index, mode)` / `set_selection({ vertices })` in [`packages/grida-svg-editor/src/core/vector-edit/session.ts`](packages/grida-svg-editor/src/core/vector-edit/session.ts).
- That session is created privately on `enter_vector_edit` in [`packages/grida-svg-editor/src/dom.ts`](packages/grida-svg-editor/src/dom.ts) (`this.vector_edit`) and is **not exposed** on the editor, the `Commands` set, or `DomSurfaceHandle`.
- The only driver today is a pointer gesture — a `select_vertex` intent fired on knob click (`handle_select_vertex` in `dom.ts`). `_do_enter_vector_edit` starts the session with `_selected_vertices = []`.

So a consumer that wants to programmatically open a path with vertex *N* already selected has no clean path; the only workaround is synthesizing a `pointerdown` at the knob's screen coordinate (fragile — requires gestures enabled + computing screen position from path coords × camera).

## Proposed API (options)

1. Optional initial selection on entry:
   ```ts
   editor.enter_content_edit(id, { vertices?: number[] /*, segments?, tangents? */ })
   ```
2. Or a dedicated command usable while in edit-content:
   ```ts
   editor.commands.select_vertices(indices: number[], mode?: "replace" | "add" | "toggle")
   ```

Either would proxy to the existing `VectorEditSession.set_selection` / `select_vertex`, so the surface stays the single owner of the session. Whichever fits the closed-command-set doctrine best.

## Use case / motivation

Programmatic setup of a vector-edit view — e.g. an agent or a host UI focusing a specific anchor, deep-linking to a vertex, or a marketing/demo surface opening a tile straight into "this anchor is selected" to make editability obvious. Came up while building the `/svg` landing page playground (live editors that open in path-edit mode); we could show the whole path in edit mode, but not highlight a specific vertex.

## Notes

Read-side (`state` exposing the current sub-selection) may be worth considering alongside the write-side, but the immediate ask is the ability to *set* it.

### PR DESCRIPTION
**Pull request by softmarshmallow**
Closes #790

## Problem

A host could put the editor into vector content-edit on a `<path>` (`enter_content_edit(id)`), but every vertex knob opened **unselected** and there was no public way to pre-select one or several. The sub-selection lived entirely in the surface-private `VectorEditSession` — neither writable nor readable through the public surface. The only workaround was synthesizing a `pointerdown` at the knob's screen coordinate (fragile).

## Approach — commands-only (Paradigm B)

The document is the only declarative extern state; interaction state is reached by replaying commands (no construction-time `initialState`). New public surface:

- **`commands.set_vector_selection(input, mode?)`** — full-triple write (`{ vertices?, segments?, tangents? }`), mid-session, **one undoable step** like a knob click. `mode` defaults to `"replace"`; `"add"` / `"toggle"` fold into the existing selection per track.
- **`enter_content_edit(id, opts?)`** — atomic entry: opens the path with the initial sub-selection in **one transition** (no unselected flash, one history step).
- **`editor.vector_subselection()` + `editor.subscribe_vector_subselection(fn)`** — designed read view on a transient channel, kept **off `state.version`** (P4 — it changes at pointer rate during marquee/lasso).
- New public types `VectorSubSelection` / `VectorSubSelectionInput`.

## Design notes for reviewers

- **Strict refusal:** an out-of-range index / tangent ref refuses the whole call (`false`, no-op) — a strict surface rejects wrong contents rather than selecting a phantom vertex.
- **Surface-owned session:** the write reaches the surface-private session through a `set_vector_subselect_driver` seam, symmetric to the existing `content_edit_driver` (not the keymap registry — keyboard isn't the ask). The read mirror pushes via `push_vector_subselection`, matching the existing pick / surface-hover channels.
- **Core-first testing:** validation + mutation live in a pure `core/vector-edit/subselection.ts` (`validate_subselection` / `apply_subselection`) so the load-bearing logic is headlessly unit-testable against a real `VectorEditSession` + `PathModel` — the DOM surface can't mount in jsdom.
- The enter-delta redo and exit-delta undo share one `reenter_vector_session` helper (no duplicated re-entry logic).
- `enter_content_edit`'s `opts` is ignored for text targets (vector-only affordance); the no-opts path is byte-identical to prior behavior.

## Demo

The `/svg` landing playground's **heart** tile now opens straight into path-edit with all anchors pre-selected via the atomic-entry form — the exact use case the issue cites. Vertices are derived from the live `PathModel` (no hardcoded indices), resolved at the page layer (host convenience, not SDK).

## Tests

- 20 new tests (`__tests__/vector-subselection.test.ts`): pure validate/apply + replace/add/toggle modes + strict refusal; command delegation; `opts` threading through `enter_content_edit`; read channel fires + no `state.version` bump.
- Updated the `_internal` seam-contract guard for the two new bridge methods.
- Manual case `TC-SVGEDITOR-VECTOR-002` for the surface-integrated path (jsdom can't mount the surface).
- Full suite green (1476 tests), `tsc --noEmit` clean, `oxlint` clean, package builds.

## Docs

README: `set_vector_selection` added to the closed Commands list; `enter_content_edit(id, opts?)` documented; new "Observation — vector sub-selection" section.

## Release Notes

* **New Features**
  * Added vector sub-selection for path editing (vertices, segments, tangents), including programmatic updates during active sessions.
  * Added observation APIs to read the current vector sub-selection and subscribe to changes; the value clears on exit.
  * Enabled fixture-driven initial pre-selection when entering path-edit mode for an atomic “enter with selection” experience.
* **Documentation**
  * Updated the vector sub-selection API reference and extended related command signatures.
* **Tests**
  * Added coverage for strict validation, refusal/no-op behavior, subscription semantics, and undo/redo correctness.

### CODE REVIEW

File: packages/grida-svg-editor/src/dom.ts [hunk 1]
```diff
@@ -4135,9 +4154,22 @@ class DomSurface implements Surface {
...
+    opts?: VectorSubSelectionInput
+  ): boolean {
     if (this.vector_edit) return false;
     if (!this._do_enter_vector_edit(id)) return false;
+    // Apply the optional initial sub-selection (gridaco/grida#790) as part of
+    // THIS entry — one history step, no intermediate empty-selection frame.
+    // `apply_subselection_to_session` self-validates against the live model;
+    // invalid opts are ignored and entry still succeeds with an empty
+    // selection (entry eligibility and sub-selection validity are separate).
+    if (opts) this.apply_subselection_to_session(opts, "replace");
```
[the remark this answers is not in the document]
Comment (softmarshmallow) [reply packages/grida-svg-editor/src/dom.ts]: Fixed in 065fd1a. `enter_vector_edit` now validates `opts` against the path **before** entering and refuses the whole call (returns `false`, no entry, no history) on an out-of-range index — matching the documented `VectorSubSelectionInput` whole-call-refusal contract and the strict `set_vector_selection`. A bad index can no longer masquerade as a successful preselect.

File: packages/grida-svg-editor/src/dom.ts [hunk 2]
```diff
@@ -4387,12 +4430,19 @@ class DomSurface implements Surface {
...
       node_id: this.vector_edit.node_id,
       vertices: this.vector_edit.selected_vertices,
       segments: this.vector_edit.selected_segments,
       tangents: this.vector_edit.selected_tangents,
-    });
+    };
+    this.hud.setVectorSelection(sel);
+    // Mirror to the editor's public read channel (gridaco/grida#790). Off the
+    // `version` stream — `subscribe_vector_subselection` consumers only.
+    this.editor_internal().push_vector_subselection(sel);
```
[the remark this answers is not in the document]
Comment (softmarshmallow) [reply packages/grida-svg-editor/src/dom.ts:4461]: Fixed in 065fd1a. `DomSurface.dispose()` now pushes `null` to the editor read channel when tearing down an active vector session. The detach/unmount path bypasses `_do_exit_vector_edit` (where the normal-exit null push lives), so this keeps `vector_subselection()` from going stale and notifies subscribers with `null` on unmount/`handle.detach()`.

File: packages/grida-svg-editor/src/dom.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (softmarshmallow) [reply packages/grida-svg-editor/src/dom.ts]: Fixed in 2c92084. `_do_enter_vector_edit` now installs the initial sub-selection BEFORE its first `sync_selection_mirror`, so atomic entry (and undo/redo re-entry) publish the final selection **once** — no transient empty frame on the read channel. (`reenter_vector_session` folded into the helper.) The invalid-opts half — silently entering with an empty selection — was already fixed in 065fd1a: out-of-range opts are validated before entering and refuse the whole call.

File: packages/grida-svg-editor/src/dom.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (softmarshmallow) [reply packages/grida-svg-editor/src/dom.ts]: Fixed in 2c92084. `sync_selection_mirror` now copies all three tracks (`[...vertices]`, `[...segments]`, `tangents.map(...)`) before publishing, so the public read channel can no longer mutate the live session arrays through the compile-time-only `readonly`.

File: packages/grida-svg-editor/src/dom.ts [hunk 3]
```diff
@@ -4547,6 +4597,49 @@ class DomSurface implements Surface {
...
+   * (see test/svg-editor-vector-subselection-api.md)
+   */
+  private apply_vector_subselection_command(
+    input: VectorSubSelectionInput,
+    mode: SelectMode = "replace"
+  ): boolean {
+    if (!this.vector_edit) return false;
+    const before = this.vector_edit.snapshot_selection();
+    if (!this.apply_subselection_to_session(input, mode)) return false;
+    this.record_vector_selection_change(before, "set vector selection");
```
[the remark this answers is not in the document]
Comment (softmarshmallow) [reply packages/grida-svg-editor/src/dom.ts:4661]: Fixed in 2c92084. `apply_vector_subselection_command` now refuses (`return false`) when `this.active_preview || this.hud.gesture().kind !== "idle"`, so a programmatic `set_vector_selection` cannot interleave a history step inside an open gesture bracket — same protection the keyboard path uses.
